instrumentation/khttp/src/main/java/io/opentelemetry/auto/instrumentation/khttp/KHttpAdvice.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.opentelemetry.auto.instrumentation.khttp;
+
+import static io.opentelemetry.auto.instrumentation.khttp.KHttpDecorator.DECORATE;
+import static io.opentelemetry.auto.instrumentation.khttp.KHttpDecorator.TRACER;
+import static io.opentelemetry.auto.instrumentation.khttp.KHttpHeadersInjectAdapter.SETTER;
+import static io.opentelemetry.context.ContextUtils.withScopedContext;
+import static io.opentelemetry.trace.Span.Kind.CLIENT;
+import static io.opentelemetry.trace.TracingContextUtils.withSpan;
+
+import io.grpc.Context;
+import io.opentelemetry.OpenTelemetry;
+import io.opentelemetry.auto.bootstrap.CallDepthThreadLocalMap;
+import io.opentelemetry.auto.instrumentation.api.SpanWithScope;
+import io.opentelemetry.trace.Span;
+import java.util.HashMap;
+import java.util.Map;
+import khttp.KHttp;
+import khttp.responses.Response;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.implementation.bytecode.assign.Assigner;
+
+public class KHttpAdvice {
+
+  @Advice.OnMethodEnter(suppress = Throwable.class)
+  public static SpanWithScope methodEnter(
+      @Advice.Argument(value = 0) String method,
+      @Advice.Argument(value = 1) String uri,
+      @Advice.Argument(
+              value = 2,
+              optional = true,
+              typing = Assigner.Typing.DYNAMIC,
+              readOnly = false)
+          Map<String, String> headers) {
+
+    final int callDepth = CallDepthThreadLocalMap.incrementCallDepth(KHttp.class);
+    if (callDepth > 0) {
+      return null;
+    }
+
+    final Span span = TRACER.spanBuilder("HTTP " + method).setSpanKind(CLIENT).startSpan();
+
+    DECORATE.afterStart(span);
+    DECORATE.onRequest(span, new RequestWrapper(method, uri));
+
+    final Context context = withSpan(span, Context.current());
+    final boolean awsClientCall = headers.containsKey("amz-sdk-invocation-id");
```

Comment:
This check shouldn't be needed unless there is an AWS SDK that has been built to use khttp as its engine. I can't find one but I could be wrong, or someone is manually doing the requests (probably unlikely due to complexity)